### PR TITLE
Add singledispatchmethod to singledispatch

### PR DESCRIPTION
**Pull request by J-M0**
The `singledispatch` package added `singledispatchmethod` in version 3.7.0. This brings the stubs inline with that update.

I don't understand why the test is failing, so I'm marking this as a draft to get help and sort it out.

### CODE REVIEW

**Comment by JelleZijlstra** [comment]
This is still failing stubtest, could you take a look?

**Comment by J-M0** [comment]
All the tests are passing now

File: stubs/singledispatch/singledispatch.pyi [hunk 1]
```diff
@@ -13,3 +13,15 @@ class _SingleDispatchCallable(Generic[_T]):
...
+class singledispatchmethod(Generic[_T]):
+    dispatcher: _SingleDispatchCallable[_T]
+    func: Callable[..., _T]
+    def __init__(self, func: Callable[..., _T]) -> None: ...
+    @overload
+    def register(self, cls: Type[Any], method: None = ...) -> Callable[[Callable[..., _T]], Callable[..., _T]]: ...
+    @overload
+    def register(self, cls: Callable[..., _T], method: None = ...) -> Callable[..., _T]: ...
+    @overload
+    def register(self, cls: Type[Any], method: Callable[..., _T]) -> Callable[..., _T]: ...
```
[the remark this answers is not in the document]
Comment (srittau) [reply stubs/singledispatch/singledispatch.pyi:26]: This question is still open. `_SingleDispatchCallable.register` only has two overloads, while this has three. They should match.

[the remark this answers is not in the document]
Comment (J-M0) [reply stubs/singledispatch/singledispatch.pyi:26]: As I said in another comment, this is how `singledispatchmethod` is implemented in the stdlib stubs, I just copied them for this backport.

https://github.com/python/typeshed/blob/ff64deb331349fa2440c28410750e24db40fbe24/stdlib/functools.pyi#L96-L106

The `__call__` method was added in this commit.

https://github.com/python/typeshed/commit/591522fb6a527318ea100d8adf99d91e662ba26f)

**Comment by itaisteinherz** [comment]
Bump :) Would be awesome to use this.

**Pull request merged**